cmd/spire-server/cli/federation/common.go

```diff
@@ -1,9 +1,160 @@
 package federation
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
 	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	"github.com/spiffe/spire/cmd/spire-server/util"
 )
 
+// FederationRelationship type is used for parsing federation relationships from file
+type federationRelationshipJSON struct {
+	TrustDomain           string          `json:"trust_domain,omitempty"`
+	BundleEndpointURL     string          `json:"bundle_endpoint_url,omitempty"`
+	BundleEndpointProfile string          `json:"bundle_endpoint_profile,omitempty"`
+	EndpointSPIFFEID      string          `json:"endpoint_spiffe_id,omitempty"`
+	Bundle                json.RawMessage `json:"bundle,omitempty"`
+	BundleFormat          string          `json:"bundle_format,omitempty"`
+	BundlePath            string
```

Comment:
Should we use conventional JSON naming here?. Also, `Bundle` stuff should be named TrustDomainBundle. We named it that way in the API to be very clear that the bundle is intended for the trust domain of the relationship and not the endpoint.
```suggestion
	TrustDomain           string          `json:"trustDomain,omitempty"`
	BundleEndpointURL     string          `json:"bundleEndpointURL,omitempty"`
	BundleEndpointProfile string          `json:"bundleEndpointProfile,omitempty"`
	EndpointSPIFFEID      string          `json:"endpointSPIFFEID,omitempty"`
	TrustDomainBundle                json.RawMessage `json:"trustDomainBundle,omitempty"`
	TrustDomainBundleFormat          string          `json:"trustDomainBundleFormat,omitempty"`
	TrustDomainBundlePath            string `json:"-"`
```